spec/ancestor_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'spec_helper'
+require 'ostruct'
+
+RSpec.describe Kitchen::Ancestor do
+
+  let(:element) { OpenStruct.new(short_type: 'page') }
+  let(:descendant) { OpenStruct.new(short_type: 'note')}
+  let(:ancestor) { described_class.new(element) }
+
+  it 'makes the type available' do
+    expect(ancestor.type).to eq element.short_type
+  end
+
+  it 'makes the element available' do
+    expect(ancestor.element).to eq element
+  end
+
+  describe '#increment_descendant_count' do
+    it 'adds 1 to the descendant count for the given type' do
+      expect { ancestor.increment_descendant_count(descendant.short_type) }.to change {ancestor.get_descendant_count(descendant.short_type)}.by(1)
```

Comment:
rubocop doesn't complain about line length b/c we have turned off that check in specs -- because we can often have long lines.  but in this case it is pretty long and we can organize it into more shorter lines:

```ruby
expect { 
  ancestor.increment_descendant_count(descendant.short_type) 
}.to change {
  ancestor.get_descendant_count(descendant.short_type)
}.by(1)
```